src/coreclr/src/jit/lsraxarch.cpp

```diff
@@ -2746,12 +2746,7 @@ int LinearScan::BuildCast(GenTreeCast* cast)
 //
 int LinearScan::BuildIndir(GenTreeIndir* indirTree)
 {
-    // If this is the rhs of a block copy (i.e. non-enregisterable struct),
-    // it has no register requirements.
-    if (indirTree->TypeGet() == TYP_STRUCT)
-    {
-        return 0;
-    }
+    assert(indirTree->TypeGet() != TYP_STRUCT); // Don't expect a struct type here.
```

Comment:
It would be more clear to say why such an indir is not expected here - `// struct typed indirs are expected to be contained` or something like that

Reply:
Thanks, done, I will change that condition soon, so it will become clearer in the next PRs as well.